src/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs

```diff
@@ -263,7 +263,11 @@ Interop.NetSecurityNative.Status status
                     tokenBytes = tokenBytes.Slice(0, length - 1);
                 }
 
+#if NETSTANDARD2_0
+                return Encoding.UTF8.GetString(tokenBytes.ToArray(), 0, tokenBytes.Length);
+#else
                 return Encoding.UTF8.GetString(tokenBytes);
+#endif
```

Comment:
Tangent from this PR (which looks fine):
Does this mean System.Data.SqlClient's nuget package on Unix depends on the .NET Core native shim?  I didn't comprehend we had such a dependency.  Having something outside the shared framework depend on the native shims is a problem.

Reply:
And does that imply that Microsoft.Data.SqlClient also does?  That'd be even worse.

Reply:
Yes, it does have a dependency on the shims. We actually discussed this at length with others several months ago.  I'll look up the issues for reference.

Reply:
See #38377 for a history of the problems with SqlClient having a dependency on these shims.  PR #38377 undid an internal API change I previously made during .NET Core 3.0 development in order to prevent breaking System.Data.SqlClient in the future.

Reply:
>And does that imply that Microsoft.Data.SqlClient also does? That'd be even worse.

I hope not! We need to prevent that and tell them to eventually use the new APIs we add from #36896

Reply:
https://github.com/dotnet/SqlClient/tree/master/src/Microsoft.Data.SqlClient/netcore/src/Common/src/Interop/Unix

@cheenamalhotra, we need to remove any such dependency from Microsoft.Data.SqlClient.  If it needs such native implementations, it needs to ship them itself as part of its nuget package.

Reply:
https://github.com/dotnet/SqlClient/issues/303